docs/how-to/debugging-guide.rst

```diff
@@ -0,0 +1,245 @@
+   details about the platform and system. Some issues to consider include:
+
+   *  Is ROCm running on a bare-metal setup, in a Docker container, in a SR-IOV virtualized
+      environment, or in some combination of these configurations? If ROCm is running in a Docker
+      container, provide the name of the Docker image.
+   *  Is the problem only seen on a specific GPU architecture?
+   *  Is it only seen on a specific system type?
+   *  Is it happening on a single node or multinode setup?
+  
+   Run the following command and collect the output:
+
+   .. code:: shell
+
+      rocm_agent_enumerator
+
+   Also collect the name of the GPU or accelerator:
+
+   .. code:: shell
+
+      rocminfo
+
+*  Run the ``rocm-smi`` command to display the system topology.
+
+   .. code:: shell
+
+      rocm-smi
+      rocm-smi --showtopo
+
+*  Determine the values of the ``PATH`` and ``LD_LIBRARY_PATH`` environment variables.
+
+   .. code:: shell
+
+      echo $PATH
+      echo $LD_LIBRARY_PATH
+
+*  Collect the HIP configuration.
+
+   .. code:: shell
+
+      /opt/rocm/bin/hipconfig --full
+
+*  Verify the network settings and setup using the ``ibv_devinfo`` command. 
+   This command displays information about the available RDMA devices available and determines 
+   whether they are installed and functioning properly.
+
+   .. code:: shell
+
+      ibv_devinfo
+
+*  Determine the BKC version, if this information is known.
+
+.. _collecting-rccl-info:
+
+Collecting RCCL information
+=============================
+
+Collect the following information about the RCCL installation and configuration.
+
+*  Run the ``ldd`` command to list any dynamic dependencies for RCCL.
+
+   .. code:: shell
+
+      ldd <specify-path-to-librccl.so>
+
+*  Determine the RCCL version. This might be the pre-packaged component in
+   ``/opt/rocm/lib`` or a version that was built from source. To verify the RCCL version,
+   enter the following command, then run either rccl-tests or an e2e application.
+
+   .. code:: shell
+
+      export NCCL_DEBUG=VERSION
+
+*  Run rccl-tests and collect the results. For information on how to build and run rccl-tests, see the
+   `rccl-tests GitHub <https://github.com/ROCm/rccl-tests/blob/develop/README.md>`_.
+
+*  Collect the RCCL logging information. Enable the debug logs, 
+   then run rccl-tests or any e2e workload to collect the logs. Use the 
+   following command to enable the logs.
+
+   .. code:: shell
+
+      export NCCL_DEBUG=INFO
+
+.. _use-rccl-replayer:
+
+Using the RCCL Replayer
+------------------------
+
+The RCCL Replayer is a debugging tool designed to analyze and replay the collective logs obtained from RCCL runs. 
+It can be useful when trying to reproduce problems, because it uses dummy data and doesn't have any dependencies 
+on non-RCCL calls. For more background on the RCCL Replayer, 
+see `the RCCL Replayer GitHub documentation <https://github.com/ROCm/rccl/tree/develop/tools/rccl_replayer>`_.
+
```

Comment:
```suggestion
The RCCL Replayer is a debugging tool designed to analyze and replay the collective logs obtained from RCCL runs. 
It can be useful when trying to reproduce problems, because it uses dummy data and doesn't have any dependencies 
on non-RCCL calls. For more information, 
see `RCCL Replayer GitHub documentation <https://github.com/ROCm/rccl/tree/develop/tools/rccl_replayer>`_.

```